File: scala_2.10.0/src/main/scala/thirtytwo/degrees/halfpipe/configuration/AbstractScalaCallback.java
package thirtytwo.degrees.halfpipe.configuration;

/**
 * User: spencergibb
 * Date: 10/17/12
 * Time: 2:00 PM
 */
@SuppressWarnings("unchecked")
public abstract class AbstractScalaCallback<C, V> implements Runnable {
    protected C config;
    protected ScalaPropertyWrapper<V> prop;

    public void setConfig(Object config) {
        this.config = (C) config;
    }

    public void setProp(Object prop) {
        if (prop instanceof ScalaPropertyWrapper) {
            this.prop = (ScalaPropertyWrapper<V>) prop;
        } else {
            System.err.println("prop is not ScalaPropertyWrapper: "+prop.getClass());
        }
    }
}

File: halfpipe-scala_2.10.0/src/main/scala/halfpipe/configuration/AbstractScalaCallback.scala
package halfpipe.configuration

/**
 * User: spencergibb
 * Date: 3/16/13
 * Time: 8:55 PM
 */
abstract class AbstractScalaCallback[C, V]() extends Runnable {
  var config: C = _
  var prop: ScalaPropertyWrapper[V] = _

  def setProp(p: Any) {
    prop = p.asInstanceOf[ScalaPropertyWrapper[V]]
  }
}
